internal/provider/resource_self_managed_worker_test.go

```diff
@@ -0,0 +1,175 @@
+	selfManagedWorkerToken      = "GzusqckarbczHoLGQ4UA25uSRGy7e7hUk4Qgz8T3NmSZF4oXBc1dKMRyH7Mr6W5u2QWggkZi1wsnfYMufb5LZJJnxEdywUxpE8RgAbVTahMhJm8oeH3nagrAKfWJkrTt8QuoqiupcTrFmJtDPZ5WBtqKSXiW3jdG8esCt7ZNKVaKV1Hq6MBXUf7duj9KfAy4Y3B31jDTzoF1uVnK1AsaLkEhgbHbuAH33L2KG5ivo7YeFeE6PknNoVavPSRSkoEpcXSfjvoPMAz9ttpNvH7jGWPwLti8r48NcVj41ftXWg"
+)
+
+var (
+	workerLedCreate = fmt.Sprintf(`
+resource "boundary_self_managed_worker" "worker_led" {
+	scope_id = "global"
+	name = "%s"
+	description = "%s"
+	worker_generated_auth_token = "%s"
+}`, selfManagedWorkerName, selfManagedWorkerDesc, selfManagedWorkerToken)
+
+	workerLedUpdate = fmt.Sprintf(`
+resource "boundary_self_managed_worker" "worker_led" {
+	scope_id = "global"
+	name = "%s"
+	description = "%s"
+	worker_generated_auth_token = "%s"
+}`, selfManagedWorkerNameUpdate, selfManagedWorkerDescUpdate, selfManagedWorkerToken)
+
+	controllerLedCreate = fmt.Sprintf(`
+resource "boundary_self_managed_worker" "controller_led" {
+	scope_id = "global"
+	name = "%s"
+	description = "%s"
+}`, selfManagedWorkerName, selfManagedWorkerDesc)
+	controllerLedUpdate = fmt.Sprintf(`
+resource "boundary_self_managed_worker" "controller_led" {
+	scope_id = "global"
+	name = "%s"
+	description = "%s"
+}`, selfManagedWorkerNameUpdate, selfManagedWorkerDescUpdate)
+)
+
+//// I'm unable to generate a worker token automatically, this will need to be run as a manual test.
+// func TestSelfManagedWorkerWorkerLed(t *testing.T) {
+// 	tc := controller.NewTestController(t, tcConfig...)
+// 	defer tc.Shutdown()
+// 	url := tc.ApiAddrs()[0]
+
+// 	var provider *schema.Provider
+// 	resource.Test(t, resource.TestCase{
+// 		ProviderFactories: providerFactories(&provider),
+// 		CheckDestroy:      testAccCheckSelfManagedWorkerResourceDestroy(t, provider),
+// 		Steps: []resource.TestStep{
+// 			{
+// 				// create
+// 				Config: testConfig(url, workerLedCreate),
+// 				Check: resource.ComposeTestCheckFunc(
+// 					testAccCheckSelfManagedWorkerResourceExists(provider, "boundary_self_managed_worker.worker_led"),
+// 					resource.TestCheckResourceAttr("boundary_self_managed_worker.worker_led", "description", selfManagedWorkerDesc),
+// 					resource.TestCheckResourceAttr("boundary_self_managed_worker.worker_led", "name", selfManagedWorkerName),
+// 				),
+// 			},
+// 			importStep("boundary_self_managed_worker.worker_led"),
+// 			{
+// 				// update
+// 				Config: testConfig(url, workerLedUpdate),
+// 				Check: resource.ComposeTestCheckFunc(
+// 					testAccCheckSelfManagedWorkerResourceExists(provider, "boundary_self_managed_worker.worker_led"),
+// 					resource.TestCheckResourceAttr("boundary_self_managed_worker.worker_led", "description", selfManagedWorkerDescUpdate),
+// 					resource.TestCheckResourceAttr("boundary_self_managed_worker.worker_led", "name", selfManagedWorkerNameUpdate),
+// 				),
+// 			},
+// 			importStep("boundary_self_managed_worker.worker_led"),
+// 		},
+// 	})
+// }
```

Comment:
I asked the team about this, and it seems like the controller led activation was added to make terraform easier, so it's probably okay for us to leave this as a manual test. However, maybe you could make it easy to test this manually by uncommenting this and making it conditional on an environment variable? Something like

```go
func TestSelfManagedWorkerWorkerLed(t *testing.T) {
    token := os.Getenv("BOUNDARY_TF_PROVIDER_TEST_WORKER_LED_TOKEN")
    if token == "" {
        t.Skip("Not running worker led activation test without worker led token present")
    }
}
```

Reply:
That makes sense, I'll make that change.